src/util.h

```diff
@@ -61,6 +61,10 @@ extern HeapLeakChecker* heap_checker;
 #include <pthread_np.h>
 #endif
 
+#define FORWARD_DECLARE_NAMESPACED(cls, ns) \
```

Comment:
Personally would insert the decls directly instead of opt for macro (kind of Bad for user to think it's fine to use the macro themselves if we intend to only use during this namespace transition and get rid of it later), but don't care much if others prefer the macro except for considering these minor suggestions:

* Renaming like `ZEEK_FORWARD_DECLARE_NAMESPACED` may be sensible since we're trying to be in the business of avoiding name clashes

* Maybe slight preference to put in `zeek-config.h` instead of `util.h` since the later comparatively pulls in more than really needed.

Reply:
I like the macro as it's pretty clearly saying what's going on. Could prefix with "__" to make clear it's internal if that helps with the concern?

Reply:
You're really not supposed to prefix things with `__` as that's generally reserved for the compiler. How about just renaming it to be prefixed with `ZEEK` and leaving a big comment above it about how it's going to be removed in 4.1 and to not use it, plus a note in NEWS about it?

Reply:
* Yes, don't prefix things with `__`

* Macro _names_ never "clearly say what's going on" because you have to look at the macro _definition_ to gain that clarity.  In this case the macro _name_ may also be more misleading on top of that since it doesn't only add a namespaced-forward-decl, it also adds a deprecated-global-alias.

* Don't think my pickiness here makes a big difference, just kind of, you know, tend to think critically about things and like to explain it, so I can take care of adding the `ZEEK_` prefix and "for internal use" comment later (when I merge in a bit).